### DISCUSSION TITLE
Invocable Action for Full Recalc via MDT (Fire Scheduled One-Off Jobs)

### DISCUSSION THREAD

**Discussion by dschibster** [jamessimone/apex-rollup#192] [Ideas]
Hi @jamessimone!

I have been noticing a pattern as of late - namely, maybe you've seen it yourself, most of my issues arose from use cases where I use Scheduled Flows in Combination with the REFRESH context. 

I think it's a great idea to leave the option, however I have also run into issues with this approach solely due to the fact that the input that Rollup supports runs into the maximum string length of a Flow (600k).

I was wondering if it might instead make sense to give users to option to not have to query data outside of a Scheduled Flow's built-in feature (which gives it batch-job-like features to be fair, but you're right that it consumes a lot of flow interviews) but instead create a Scheduled Flow that fires an Invocable Action with the **DeveloperName of the Rollup__mdt record you want to run, and have the rest be done by the FullRecalcProcessor like in the App**.

Does this sound like a good idea? I'd probably try to make a fork and create the Action myself if you think it's sensible.

**Comment by jamessimone** [comment]
@dschibster I think that sounds great - one piece of feedback, though, is that I think it makes more sense to pass in a comma delimited string, or a text collection, so that you can query for multiple Rollup__mdt at a time.

**Comment by jamessimone** [comment]
I can also create another InvocationPoint enum for you in this version (`SCHEDULED_FLOW`), which is the only other argument you’d need to pass to the `@AuraEnabled` method called by the app

**Reply by dschibster** [reply]
Sounds great!

**Reply by jamessimone** [reply]
@dschibster if you are looking to complete this work this week, please use `v1.3.9` as the base branch when making the PR. Thank you!

**Reply by dschibster** [reply]
I'm planning to! Here's a quick link to the commit where I created the Invocable: https://github.com/dschibster/apex-rollup/commit/e1c05f707eaf8e298cdf4f5e5cc672765b39bfcb 

Still needs a test class though.

**Reply by jamessimone** [reply]
Left two small comments, looks great otherwise!
